File: Testing/Stdlib Test Code/test_array_methods.js

function array_sum(arr, n) {
    let total = 0;
    let i = 0;
    while ((i < n)) {
        total = (total + arr[i]);
        i = (i + 1);
    }
    return total;
}

function array_contains(arr, n, val) {
    let i = 0;
    while ((i < n)) {
        if ((arr[i] === val)) {
            return true;
        }
        i = (i + 1);
    }
    return false;
}

function main() {
    let nums = [3, 1, 4, 1, 5];
    let total = nums.reduce((a, b) => a + b, 0);
    console.log(total);
    let found = nums.includes(4);
    console.log(found);
    let nope = nums.includes(9);
    console.log(nope);
    return 0;
}

function __ul_sort(arr, n) { const sub = arr.slice(0, n).sort((a, b) => a < b ? -1 : a > b ? 1 : 0); for (let i = 0; i < n; i++) arr[i] = sub[i]; }

main();

File: Testing/Stdlib Test Code/test_array_methods.py
from dataclasses import dataclass, field

def __ul_sort(arr, n): arr[:n] = sorted(arr[:n])

class __ul_Exception(Exception):
    def __init__(self, type_name, message):
        self.type_name = type_name
        self.message = message
        super().__init__(message)


def array_sum(arr, n):
    total = 0
    i = 0
    while (i < n):
        total = (total + arr[i])
        i = (i + 1)
    return total

def array_contains(arr, n, val):
    i = 0
    while (i < n):
        if (arr[i] == val):
            return True
        i = (i + 1)
    return False

def main():
    nums = [3, 1, 4, 1, 5]
    total = sum(nums)
    print(total)
    found = (4 in nums)
    print(found)
    nope = (9 in nums)
    print(nope)
    return 0


if __name__ == "__main__":
    main()
